src/platform/cannonlake/cannonlake.x.in

```diff
@@ -443,7 +443,7 @@ SECTIONS
     . = . + HEAP_RUNTIME_SIZE;
     _runtime_heap_end = ABSOLUTE(.);
 
-    . = ALIGN (32);
+    . = ALIGN (XCHAL_DCACHE_LINESIZE);
```

Comment:
Please use PLATFORM_DCACHE_ALIGN.

Reply:
@tlauda currently not possible, we cannot include sof/platform.h to linker and the value of PLATFORM_DCACHE_ALIGN can be "sizeof(uint32_t)" which will not work in linker.

If we want to use it then we need to move it to another file and remove sizeof possibility

Reply:
This is platform specific linker script, not a universal one, so I'm fine with some HEAP_BUF_ALIGNMENT define in some common cAVS header file included by the linker script and also in another place followed by some C_ASSERT(0 == (HEAP_BUF_ALIGNMENT % PLATFORM_DCACHE_ALIGN)).

Reply:
@tlauda @mmaka1 added HEAP_BUF_ALIGNMENT with static_assert for it.
Done in cAVS memory.h and memory.c, please check if this is a correct place for them